NPE when encrypting fields for SQLServer databases using debezium

I am using debezium to do a CDC from SQLServer to kafka, and as per the business needs, some of the columns must be encrypted.

Here is the snippet of the connector json file

```
{
"name": "live.sql.users",
...
        "transforms.unwrap.delete.handling.mode": "drop",
        "transforms": "unwrap,cipher",
        "predicates.isTombstone.type": "org.apache.kafka.connect.transforms.predicates.RecordIsTombstone",
        "transforms.unwrap.drop.tombstones": "false",
        "transforms.unwrap.type": "io.debezium.transforms.ExtractNewRecordState",
        "transforms.cipher.predicate": "isTombstone",
        "transforms.cipher.negate": "true",
        "transforms.cipher.cipher_data_keys": "[ { \"identifier\": \"my-key\", \"material\": { \"primaryKeyId\": 1000000001, \"key\": [ { \"keyData\": { \"typeUrl\": \"type.googleapis.com/google.crypto.tink.AesGcmKey\", \"value\": \"GhDLeulEJRDC8/19NMUXqw2jK\", \"keyMaterialType\": \"SYMMETRIC\" }, \"status\": \"ENABLED\", \"keyId\": 2000000002, \"outputPrefixType\": \"TINK\" } ] } } ]",
        "transforms.cipher.type": "com.github.hpgrahsl.kafka.connect.transforms.kryptonite.CipherField$Value",
        "transforms.cipher.cipher_mode": "ENCRYPT",
        "predicates": "isTombstone",
        "transforms.cipher.field_config": "[{\"name\":\"Password\"},{\"name\":\"MobNumber\"}, {\"name\":\"UserName\"}]",
        "transforms.cipher.cipher_data_key_identifier": "my-key"
...
}
```

and when I applied it, after few seconds I got the below error, when I call the `/connectors/<connector_name>/status` api
```
org.apache.kafka.connect.errors.ConnectException: Tolerance exceeded in error handler\n\t
at org.apache.kafka.connect.runtime.errors.RetryWithToleranceOperator.execAndHandleError(RetryWithToleranceOperator.java:206)\n\t
at org.apache.kafka.connect.runtime.errors.RetryWithToleranceOperator.execute(RetryWithToleranceOperator.java:132)\n\t
at org.apache.kafka.connect.runtime.TransformationChain.apply(TransformationChain.java:50)\n\t
at org.apache.kafka.connect.runtime.WorkerSourceTask.sendRecords(WorkerSourceTask.java:346)\n\t
at org.apache.kafka.connect.runtime.WorkerSourceTask.execute(WorkerSourceTask.java:261)\n\t
at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:191)\n\t
at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:240)\n\t
at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)\n\t
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)\n\t
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)\n\t
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)\n\t
at java.base/java.lang.Thread.run(Thread.java:829)\nCaused by: org.apache.kafka.connect.errors.DataException: error: ENCRYPT of field path 'UserName' having data 'deleted605' failed unexpectedly\n\t
at com.github.hpgrahsl.kafka.connect.transforms.kryptonite.RecordHandler.processField(RecordHandler.java:90)\n\t
at com.github.hpgrahsl.kafka.connect.transforms.kryptonite.SchemaawareRecordHandler.lambda$matchFields$0(SchemaawareRecordHandler.java:73)\n\t
at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)\n\t
at java.base/java.util.Collections$UnmodifiableCollection.forEach(Collections.java:1085)\n\t
at com.github.hpgrahsl.kafka.connect.transforms.kryptonite.SchemaawareRecordHandler.matchFields(SchemaawareRecordHandler.java:50)\n\t
at com.github.hpgrahsl.kafka.connect.transforms.kryptonite.CipherField.processWithSchema(CipherField.java:163)\n\t
at com.github.hpgrahsl.kafka.connect.transforms.kryptonite.CipherField.apply(CipherField.java:140)\n\t
at org.apache.kafka.connect.runtime.PredicatedTransformation.apply(PredicatedTransformation.java:56)\n\t
at org.apache.kafka.connect.runtime.TransformationChain.lambda$apply$0(TransformationChain.java:50)\n\t
at org.apache.kafka.connect.runtime.errors.RetryWithToleranceOperator.execAndRetry(RetryWithToleranceOperator.java:156)\n\t
at org.apache.kafka.connect.runtime.errors.RetryWithToleranceOperator.execAndHandleError(RetryWithToleranceOperator.java:190)\n\t
... 11 more\nCaused by: java.lang.NullPointerException\n\t
at com.esotericsoftware.kryo.util.DefaultGenerics.nextGenericTypes(DefaultGenerics.java:77)\n\t
at com.esotericsoftware.kryo.serializers.FieldSerializer.pushTypeVariables(FieldSerializer.java:144)\n\t
at com.esotericsoftware.kryo.serializers.FieldSerializer.write(FieldSerializer.java:102)\n\t
at com.esotericsoftware.kryo.Kryo.writeObject(Kryo.java:627)\n\t
at com.github.hpgrahsl.kafka.connect.transforms.kryptonite.RecordHandler.processField(RecordHandler.java:75)\n\t
... 21 more\n
```

Knowing that, the same configs working with other connectors with no problems

--- a/connect-transform-kryptonite/pom.xml
+++ b/connect-transform-kryptonite/pom.xml
@@ -22,15 +22,15 @@
   <parent>
     <groupId>com.github.hpgrahsl</groupId>
     <artifactId>kryptonite-for-kafka</artifactId>
-    <version>0.4.0</version>
+    <version>0.4.1</version>
     <relativePath>../pom.xml</relativePath>
   </parent>
   <artifactId>connect-transform-kryptonite</artifactId>
-  <version>0.4.0</version>
+  <version>0.4.1</version>
   <packaging>jar</packaging>
 
   <properties>
-    <kryptonite.version>0.4.0</kryptonite.version>
+    <kryptonite.version>0.4.1</kryptonite.version>
     <kafka.version>3.3.1</kafka.version>
     <jackson.version>2.14.1</jackson.version>
     <logback.version>1.2.11</logback.version>

--- a/funqy-http-kryptonite/pom.xml
+++ b/funqy-http-kryptonite/pom.xml
@@ -5,19 +5,19 @@
   <parent>
     <groupId>com.github.hpgrahsl</groupId>
     <artifactId>kryptonite-for-kafka</artifactId>
-    <version>0.4.0</version>
+    <version>0.4.1</version>
   </parent>
   <groupId>com.github.hpgrahsl</groupId>
   <artifactId>funqy-http-kryptonite</artifactId>
-  <version>0.1.0</version>
+  <version>0.1.1</version>
   <name>funqy-http-kryptonite</name>
   <properties>
     <compiler-plugin.version>3.10.1</compiler-plugin.version>
-    <google.tink.version>1.6.1</google.tink.version>
+    <google.tink.version>1.7.0</google.tink.version>
     <kafka.version>3.3.1</kafka.version>
     <kryo.serializers>0.45</kryo.serializers>
-    <kryo.version>5.3.0</kryo.version>
-    <kryptonite.version>0.4.0</kryptonite.version>
+    <kryo.version>5.4.0</kryo.version>
+    <kryptonite.version>0.4.1</kryptonite.version>
     <maven.compiler.release>11</maven.compiler.release>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>

--- a/kryptonite/pom.xml
+++ b/kryptonite/pom.xml
@@ -22,16 +22,16 @@
   <parent>
     <groupId>com.github.hpgrahsl</groupId>
     <artifactId>kryptonite-for-kafka</artifactId>
-    <version>0.4.0</version>
+    <version>0.4.1</version>
     <relativePath>../pom.xml</relativePath>
   </parent>
   <artifactId>kryptonite</artifactId>
-  <version>0.4.0</version>
+  <version>0.4.1</version>
   <packaging>jar</packaging>
 
   <properties>
     <google.tink.version>1.7.0</google.tink.version>
-    <kryo.version>5.3.0</kryo.version>
+    <kryo.version>5.4.0</kryo.version>
     <kryo.serializers>0.45</kryo.serializers>
     <az.keyvault.version>4.5.2</az.keyvault.version>
     <az.identity.version>1.7.1</az.identity.version>

--- a/kryptonite/src/main/java/com/github/hpgrahsl/kryptonite/serdes/KryoInstance.java
+++ b/kryptonite/src/main/java/com/github/hpgrahsl/kryptonite/serdes/KryoInstance.java
@@ -35,53 +35,56 @@ import java.util.*;
 
 public class KryoInstance {
 
-  private static final Kryo KRYO = new Kryo();
-
-  static {
-    try {
-      KRYO.setWarnUnregisteredClasses(true);
-      KRYO.setRegistrationRequired(false);
-      KRYO.setInstantiatorStrategy(new DefaultInstantiatorStrategy(new StdInstantiatorStrategy()));
-      //NOTE: pre-register kryptonite for kafka specific classes and if applicable the necessary custom serializers
-      KRYO.register(FieldMetaData.class);
-      KRYO.register(PayloadMetaData.class);
-      KRYO.register(EncryptedField.class);
-      KRYO.register(Struct.class).setSerializer(new KryoSerdeProcessor.StructSerializer());
-      KRYO.register(Schema.class).setSerializer(new KryoSerdeProcessor.SchemaSerializer());
-      KRYO.register(Schema.Type.class);
-      //NOTE: pre-registering a couple of commonly found classes
-      //      in the context of kafka connect and ksqlDB
-      KRYO.register(Object.class);
-      KRYO.register(byte[].class);
-      KRYO.register(BigDecimal.class);
-      KRYO.register(List.class);
-      KRYO.register(ArrayList.class);
-      KRYO.register(LinkedList.class);
-      KRYO.register(Map.class);
-      KRYO.register(HashMap.class);
-      KRYO.register(LinkedHashMap.class);
-      KRYO.register(Set.class);
-      KRYO.register(HashSet.class);
-      KRYO.register(LinkedHashSet.class);
-      KRYO.register(Date.class);
-      KRYO.register(Time.class);
-      KRYO.register(Timestamp.class);
-      KRYO.register(Class.forName("java.util.Arrays$ArrayList"));
-      KRYO.register(Class.forName("java.util.ImmutableCollections$ListN"));
-      KRYO.register(Class.forName("java.util.ImmutableCollections$List12"));
-      KRYO.register(Class.forName("java.util.ImmutableCollections$SetN"));
-      KRYO.register(Class.forName("java.util.ImmutableCollections$Map1"));
-      KRYO.register(Class.forName("java.util.ImmutableCollections$MapN"));
-      //NOTE: kryo community serializers for other specific collection types
-      UnmodifiableCollectionsSerializer.registerSerializers(KRYO);
-      SynchronizedCollectionsSerializer.registerSerializers(KRYO);
-    } catch (ClassNotFoundException e) {
-      throw new RuntimeException(e);
-    }
-  }
+  private static final ThreadLocal<Kryo> KRYOS = new ThreadLocal<Kryo>() {
+    protected Kryo initialValue() {
+      Kryo kryo = new Kryo();
+      try {
+        kryo.setWarnUnregisteredClasses(true);
+        kryo.setRegistrationRequired(false);
+        kryo.setInstantiatorStrategy(new DefaultInstantiatorStrategy(new StdInstantiatorStrategy()));
+        // NOTE: pre-register kryptonite for kafka specific classes and if applicable
+        // the necessary custom serializers
+        kryo.register(FieldMetaData.class);
+        kryo.register(PayloadMetaData.class);
+        kryo.register(EncryptedField.class);
+        kryo.register(Struct.class).setSerializer(new KryoSerdeProcessor.StructSerializer());
+        kryo.register(Schema.class).setSerializer(new KryoSerdeProcessor.SchemaSerializer());
+        kryo.register(Schema.Type.class);
+        // NOTE: pre-registering a couple of commonly found classes
+        // in the context of kafka connect and ksqlDB
+        kryo.register(Object.class);
+        kryo.register(byte[].class);
+        kryo.register(BigDecimal.class);
+        kryo.register(List.class);
+        kryo.register(ArrayList.class);
+        kryo.register(LinkedList.class);
+        kryo.register(Map.class);
+        kryo.register(HashMap.class);
+        kryo.register(LinkedHashMap.class);
+        kryo.register(Set.class);
+        kryo.register(HashSet.class);
+        kryo.register(LinkedHashSet.class);
+        kryo.register(Date.class);
+        kryo.register(Time.class);
+        kryo.register(Timestamp.class);
+        kryo.register(Class.forName("java.util.Arrays$ArrayList"));
+        kryo.register(Class.forName("java.util.ImmutableCollections$ListN"));
+        kryo.register(Class.forName("java.util.ImmutableCollections$List12"));
+        kryo.register(Class.forName("java.util.ImmutableCollections$SetN"));
+        kryo.register(Class.forName("java.util.ImmutableCollections$Map1"));
+        kryo.register(Class.forName("java.util.ImmutableCollections$MapN"));
+        // NOTE: kryo community serializers for other specific collection types
+        UnmodifiableCollectionsSerializer.registerSerializers(kryo);
+        SynchronizedCollectionsSerializer.registerSerializers(kryo);
+        return kryo;
+      } catch (ClassNotFoundException e) {
+        throw new RuntimeException(e);
+      }
+    };
+  };
 
   public static Kryo get() {
-    return KRYO;
+    return KRYOS.get();
   }
 
 }

--- a/ksqldb-udfs-kryptonite/pom.xml
+++ b/ksqldb-udfs-kryptonite/pom.xml
@@ -23,15 +23,15 @@
     <parent>
         <groupId>com.github.hpgrahsl</groupId>
         <artifactId>kryptonite-for-kafka</artifactId>
-        <version>0.4.0</version>
+        <version>0.4.1</version>
         <relativePath>../pom.xml</relativePath>
     </parent>
     <artifactId>ksqldb-udfs-kryptonite</artifactId>
-    <version>0.2.0</version>
+    <version>0.2.1</version>
     <packaging>jar</packaging>
 
     <properties>
-        <kryptonite.version>0.4.0</kryptonite.version>
+        <kryptonite.version>0.4.1</kryptonite.version>
         <ksql.version>7.3.0</ksql.version>
         <jackson.version>2.14.1</jackson.version>
         <logback.version>1.2.11</logback.version>

--- a/pom.xml
+++ b/pom.xml
@@ -20,7 +20,7 @@
   <modelVersion>4.0.0</modelVersion>
   <groupId>com.github.hpgrahsl</groupId>
   <artifactId>kryptonite-for-kafka</artifactId>
-  <version>0.4.0</version>
+  <version>0.4.1</version>
   <packaging>pom</packaging>
 
   <modules>

